utils/ratelimits.py

```diff
@@ -0,0 +1,101 @@
+from datetime import datetime, timedelta
+import json
+
+from flask import abort, request
+
+
+class KeyExpired(KeyError):
+    pass
+
+
+class RatelimitCache(object):
+    def __init__(self, expire_time=timedelta(0, 1, 0)):
+        self.expire_time = expire_time
+        self.cache = {}
+
+    def __getitem__(self, item):
+        c = self.cache[item]
+
+        now = datetime.now()
+        if now - c['timestamp'] < c['expire_time']:
+            return c['data']
+
+        del self.cache[item]
+        raise KeyExpired(item)
+
+    def get(self, item):
+        c = self.cache[item]
+
+        now = datetime.now()
+        if now - c['timestamp'] < c['expire_time']:
+            return c['data']
+
+        del self.cache[item]
+        raise KeyExpired(item)
+
+    def __contains__(self, item):
+        try:
+            self[item]
+            return True
+        except KeyError:
+            return False
+
+    def __setitem__(self, key, value):
+        self.cache[key] = {
+            'data': value,
+            'timestamp': datetime.now(),
+            'expire_time': self.expire_time
+        }
+
+    def set(self, key, value):
+        self.cache[key] = {
+            'data': value,
+            'timestamp': datetime.now(),
+            'expire_time': self.expire_time
+        }
+
+
+cache = RatelimitCache()
+
+
+def get_config():
+    try:
+        with open('config.json') as config:
```

Comment:
Not sure I like the idea of constant file IO. Perhaps this could be loaded once and stored in memory.

Reply:
Resolved. Do keep in mind that your get_all_keys() function also has the same IO